docs/source/array-slicing.rst

```diff
@@ -46,4 +48,49 @@ we are also a bit wasteful in that we still need to manipulate the Dask graph
 with a million or so tasks in it.  This can cause an interactive overhead of a
 second or two. 
 
-But generally, slicing works well.
+Slicing with concrete indexers (a list of integers, say) has a couple of possible
+failure modes that are worth mentioning. First, when you're indexing a chunked
+axis, Dask will typically "match" the chunking on the output.
+
+.. code-block:: python
+
+   # Array of ones, chunked along axis 0
+   >>> a = da.ones((4, 10000, 10000), chunks=(1, -1, -1))
+
+If we slice that with a *sorted* sequence of integers, Dask will return one chunk
+per intput chunk
+
+   >>> a[[0, 1], :, :]
+   dask.array<getitem, shape=(2, 10000, 10000), dtype=float64, chunksize=(1, 10000, 10000), chunktype=numpy.ndarray>
+
+But what about repeated indices? Dask continues to return one chunk per input chunk,
+but if you have many repetitions from the same input chunk, your output chunk could
+be much larger.
+
+.. code-block:: python
+
+   >>> a[[0] * 15, :, :]
+   PerformanceWarning: Slicing is producing a large chunk. To accept the large
+   chunk and silence this warning, set the option
+       >>> with dask.config.set(**{'array.slicing.split_large_chunks': False}):
+       ...     array[indexer]
+   
+   To avoid creating the large chunks, set the option
+       >>> with dask.config.set(**{'array.slicing.split_large_chunks': True}):
+       ...     array[indexer]
+   dask.array<getitem, shape=(15, 10000, 10000), dtype=float64, chunksize=(15, 10000, 10000), chunktype=numpy.ndarray>
+
+Previously we had a chunksize of ``1`` along the first dimension. But we've
+selected 15 elements from that first chunk, producing a large output chunk.
+
+Dask warns when indexing like this produces a chunk that's 10x larger
```

Comment:
Maybe 2x or 5x would be better?  10x seems like a large chunk to me.

Reply:
Thank you for the doc by the way. This was very informative to me (I only sort of tracked the previous conversation).  I'm now curious to learn about situations where we don't want to split up these chunks, but I suppose that that's already covered in conversation on the issue, and I can go and read there.

Reply:
Changed to 5x, it's pretty arbitrary.

> I'm now curious to learn about situations where we don't want to split up these chunks

It sounds like there might be a few issues. For xarray, some operations in a Dataset require uniform chunks. Since the chunk splitting logic depends on the `itemsize` of the array, you could end up with a call where

```python
ds = load_dasets()  # int64 and int32 arrays, uniform chunks
ds.sort_by("column")  # eventually calls Array.__getitem__, splits according to itemsize -> inconsistent chunks
```